proposals/nft.storage-chunked-car-uploads.md

```diff
@@ -0,0 +1,107 @@
+# nft.storage chunked CAR uploads
+
+Authors: [@alanshaw](https://github.com/alanshaw) [@vasco-santos](https://github.com/vasco-santos) [@olizilla](https://github.com/olizilla)
+
+## What is the problem this project solves?
+_Describe the status quo, including any relevant context on the problem you're seeing that this project should solve. Who is the user you're solving for, and why do they care about this problem? Wherever possible, include pain points or problems that you've seen users experience to help motivate why solving this problem works towards top-line objectives._ 
+
+NFT.storage has a hard limit of 100MB for uploads - a limit imposed by Cloudflare Workers (the server side technology that runs the site).
+
+This project proposes a way to work around this limit by sending _multiple_ requests of <100MB.
+
+## Impact
+_What goals/OKRs are being addressed (for w3dt, a specific program, etc.)? Why is this project important? What do we get with this project that we can't get without it?_
+
+Allows NFT assets over 100MB to be reliably uploaded to nft.strorage.
+
+This has been requested by multiple marketplaces and is necessary for NFTs that contain video content.
+
+## The idea
+_Describe the proposed project solution, at a very high level. Stay at the level of the high-level requirements. Diagrams and interface descriptions can be useful, if you have any that help clarify and explain the idea._
+
+1. Generate a CAR file for the content.
+
+    This has the added benefit of allowing the creator to know the CID up front, allowing them to mint an NFT on a blockchain and upload the content to the nft.storage in parallel.
+
+    Note: the CAR file must be generated by the NFT creator using a tool such as [ipfs-car](https://github.com/vasco-santos/ipfs-car).
+    
+2. Split the CAR file into chunks of <100MB.
+
+    The splitting process does not attempt to understand the DAG, it simply reads the blocks until it has reached 100MB and then starts creating a new CAR with the remaining blocks.
+    
+    Each CAR has a "chunk root" node which references all the blocks and roots (only the first chunk) in the CAR.
```

Comment:
Is the chunk root listed as a root in the CAR file? It seems like there are some (very solvable) edge cases here around the maximum size that the chunker would need to be aware of. Some examples include:

1) If there are lots of blocks that need to go under the root needing to build a HAMT for the chunk root. Not counting the size of the chunk root related data could push you over the limit.
2) If there are large identity CIDs then you could run into trouble, particularly with the chunk root because you'll be duplicating the data.

Reply:
Side note: If we were willing to pin the roots first then the complexity around the edge cases would go away and you could literally just grab 100MB worth of blocks (minus the header size).